sdk/core/azure-core/src/uuid.cpp

```diff
@@ -67,8 +67,15 @@ namespace Azure { namespace Core {
       std::memcpy(uuid + i, &x, 4);
     }
 
-    // SetVariant to ReservedRFC4122
-    uuid[8] = (uuid[8] | ReservedRFC4122) & 0x7F;
```

Comment:
This line isn’t doing what is says it is supposed to be doing, from my reading of it (unless I got the endianness/bit order wrong).
https://www.rfc-editor.org/rfc/rfc4122.html#section-4.1.1

Incorrect: (X | 0100 000) & (0111 1111)

It is setting the high bits to `01xx xxxx` instead of what it should be `10xx xxxx`.

Correct: (X | 1000 000) & (0011 1111)

Reply:
Confirmed with .NET and GoLang's UUID implementations, they only produce values that have 8,9,A|a,B|b for the 17th digit in the form `8-4-4-4-12`

```golang
package main

import (
	"fmt"

	"github.com/google/uuid"
)

func main() {
	for i := 0; i < 10; i++ {
		id := uuid.New()
		fmt.Println(id.String())
	}
}
```
```C#
using System;
					
public class Program
{
	public static void Main()
	{
		for (int i = 0; i < 256; i++)
		{
			Console.WriteLine(Guid.NewGuid());
		}

	}
}
```

```text
eb5d6e43-8be3-4d83-a584-a9033c68788b
fe813296-b8bb-4565-9c7a-697d18dffc25
...
869f397c-27b1-4ec9-ae35-f4ce244d3fe7
31321838-d701-4280-8c5a-7affad82086b
927df9e6-26cb-4b0e-8291-be207e012043
49cd84e0-8cda-462c-913f-e352740d5520
```

Reply:
Funny - I was thinking the same thing when I was modifying this code. Good catch.